### PR TITLE
Add batch of fast K6 tests

### PR DESCRIPTION
**Pull request by Piotrk39**
AB#38067 

## Describe your changes

Adds __ENV parameter to speed up the K6 execution speed so they can run every PR.

## Checklist before requesting a code review

- [ ] I have performed a self-review of my code
- [ ] Adding tests is unnecessary/irrelevant
- [ ] The changes do not touch the UI/UX
- [ ] I have made sure that all automated checks pass before requesting a review
- [ ] I do not need any deviation from our PR guidelines

## Portal preview-deployment

This PR does not have any preview deployments yet.

### CODE REVIEW

File: k6/models/payments.js [hunk 1]
```diff
@@ -4,7 +4,22 @@ import http from 'k6/http';
 import config from './config.js';
 const { baseUrl } = config;
 
-export default class paymentsModel {
+export default class PaymentsModel {
+  verifyPaymentDryRun(programId, amount) {
```
Comment (RubenGeo) [inline k6/models/payments.js:8]: I am not sure if I understand it correctly. But should you not also use verifyPaymentDryRunUntilSuccess here?

**Review by RubenGeo — CHANGES_REQUESTED** [review]
Great changes!  Hope this also gets rid of the flakeyness. Just some small comments on the code.

For the rest I think it would also be nice if you add a small section to the readme of how to run these tests locally and than set your own duplicate number. :)

File: k6/tests/findDuplicates100kRegistrations.js [hunk 2]
```diff
@@ -27,7 +42,19 @@ function checkAndFail(response, checks) {
...
+}
+
 export default function () {
+  // Wait for the service to restart and become available
+  let serviceUp = false;
+  while (!serviceUp) {
+    sleep(1); // wait for 1 second before retrying
+    serviceUp = isServiceUp();
+  }
+  // reset DB
```
[the remark this answers is not in the document]
Comment (Piotrk39) [reply k6/tests/findDuplicates100kRegistrations.js:57]: Yes without this after the finish of previous test and the reset of the other some APIs are still unavailable and at this step test needs to implicitly wait for the service to be up. Without it it fails every time. Tests in the fast run are executed a bit differently than in cronjob.

File: k6/README.md [hunk 3]
```diff
@@ -1,67 +1,111 @@
...
 ```
 
-Make sure ENV-variable `EXTERNAL_121_SERVICE_URL` is set to `http://localhost:3000` (and not to some `ngrok` address)
+**To:**
 
-**To run the tests:**
+```javascript
+baseUrl: __ENV.EXTERNAL_121_SERVICE_URL || 'http://localhost:3000/',
+```
+
```
Comment (RubenGeo) [inline k6/README.md:62]: Instead of this you could change config.js to: 

```
/* global __ENV */
export default {
  baseUrl: __ENV.EXTERNAL_121_SERVICE_URL
    ? __ENV.EXTERNAL_121_SERVICE_URL.replace(/\/+$/, '') + '/' // This ensures there's exactly one trailing slash
    : 'http://localhost:3000/',
  credentials: {
    username: __ENV.USERCONFIG_121_SERVICE_EMAIL_ADMIN,
    password: __ENV.USERCONFIG_121_SERVICE_PASSWORD_ADMIN,
  },
};
```

So you always trim the / from the url and you should never end up with 2 slashes

File: k6/README.md [hunk 4]
```diff
@@ -1,67 +1,111 @@
...
 
-Then:
+#### Adjusting Test Load
 
-```shell
-npm run test -- tests/<name of the test>.js
+For faster CI execution, tests use:
+
+```javascript
+const duplicateNumber = parseInt(__ENV.DUPLICATE_NUMBER || '5');
```
Comment (RubenGeo) [inline k6/README.md:75]:
```suggestion
For faster CI execution, tests use:

```shell
DUPLICATE_NUMBER=5 npm run test -- tests/<name-of-the-test>.js
```

One of the nice things of using ENV variables is that you do not need to make code changes to run code with a different configuration. 

So this way you use the env variable locally the same way as you would use it in the ci pipeline

**Review by RubenGeo — CHANGES_REQUESTED** [review]
See 2 comments :)

File: k6/README.md (hunk 3 above)
Comment (Piotrk39) [reply k6/README.md:62]: I changed it as per suggestion
